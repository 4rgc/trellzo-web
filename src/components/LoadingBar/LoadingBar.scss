.loadbar-container {
    border: solid 1px #1e1e1e;
    border-radius: 10px;
    background-color: #5f5f5f;
    overflow: hidden;
}

.loadbar-inner {
    transition: width ease 0.4s;
    height: 100%;
}

.loadbar-small {
    height: 10px;
}

.loadbar-large {
    height: 15px;
}

.loadbar-green {
    background-color: #66dd66;
}

.loadbar-blue {
    background-color: #5577ff;
}
