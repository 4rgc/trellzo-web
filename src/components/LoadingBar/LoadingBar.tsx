import { FC } from 'react';

import './LoadingBar.scss';

export interface LoadingBarProps extends React.ComponentPropsWithoutRef<'div'> {
	state?: number;
	size?: 'small' | 'large';
	color?: string;
}

const LoadingBar: FC<LoadingBarProps> = ({
	state = 0,
	size = 'small',
	color = 'green',
	className,
}) => {
	return (
		<div className={`loadbar-container loadbar-${size} ${className}`}>
			<div
				className={`loadbar-inner`}
				style={{ width: `${state}%`, backgroundColor: `${color}` }}
			/>
		</div>
	);
};

export default LoadingBar;
