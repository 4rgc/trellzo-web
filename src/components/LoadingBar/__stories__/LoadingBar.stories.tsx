import { Meta, StoryFn } from '@storybook/react';
import LoadingBar from '../LoadingBar';

export default {
	title: 'Components/LoadingBar',
	component: LoadingBar,
} as Meta<typeof LoadingBar>;

const Template: StoryFn<typeof LoadingBar> = (args) => (
	<div style={{ width: '120px' }}>
		<LoadingBar {...args} />
	</div>
);

export const Default = Template.bind({});

export const Large = Template.bind({});
Large.args = { size: 'large' };

export const Blue = Template.bind({});
Blue.args = { color: 'blue', state: 100 };

export const Green = Template.bind({});
Green.args = { color: 'green', state: 100 };

export const HalfFull = Template.bind({});
HalfFull.args = { state: 50 };
