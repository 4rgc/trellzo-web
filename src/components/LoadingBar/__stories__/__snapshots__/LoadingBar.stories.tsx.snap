// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/LoadingBar Blue smoke-test 1`] = `
<div style="width: 120px;">
  <div class="loadbar-container loadbar-small undefined">
    <div class="loadbar-inner"
         style="width: 100%; background-color: blue;"
    >
    </div>
  </div>
</div>
`;

exports[`Components/LoadingBar Default smoke-test 1`] = `
<div style="width: 120px;">
  <div class="loadbar-container loadbar-small undefined">
    <div class="loadbar-inner"
         style="width: 0%; background-color: green;"
    >
    </div>
  </div>
</div>
`;

exports[`Components/LoadingBar Green smoke-test 1`] = `
<div style="width: 120px;">
  <div class="loadbar-container loadbar-small undefined">
    <div class="loadbar-inner"
         style="width: 100%; background-color: green;"
    >
    </div>
  </div>
</div>
`;

exports[`Components/LoadingBar HalfFull smoke-test 1`] = `
<div style="width: 120px;">
  <div class="loadbar-container loadbar-small undefined">
    <div class="loadbar-inner"
         style="width: 50%; background-color: green;"
    >
    </div>
  </div>
</div>
`;

exports[`Components/LoadingBar Large smoke-test 1`] = `
<div style="width: 120px;">
  <div class="loadbar-container loadbar-large undefined">
    <div class="loadbar-inner"
         style="width: 0%; background-color: green;"
    >
    </div>
  </div>
</div>
`;
